src/CentreonRemote/Application/Clapi/CentreonRemoteServer.php

```diff
@@ -47,65 +53,65 @@ public function enableRemote(string $string_ip)
         );
 
         /* Check CLAPI */
-        $options = explode(';', $string_ip);
+        $options = explode(';', $parametersString);
 
         if (count($options) === 6) {
-            $string_ip = $options[0];
+            $urlString = $options[0];
             $noCheckCertificate = $options[1];
             $data['remoteHttpMethod'] = $options[2];
             $data['remoteHttpPort'] = $options[3];
             $data['remoteNoCheckCertificate'] = $options[4];
             $noProxy = $options[5];
         } elseif (count($options) > 1) {
-            echo "6 arguments are needed, please check your arguments.";
+            echo "Expecting 6 parameters, received " . count($options) . "\n";
             return 1;
         }
 
         /* Extract host from URI */
-        $aIPMaster = array();
-        $pattern_extract_host = '/^[a-z][a-z0-9+\-.]*:\/\/([a-z0-9\-._~%!$&\'()*+,;=]+@)?([a-z0-9\-._~%]+|\[[a-z0-9\-._~%!$&\'()*+,;=:]+\])/';
-        $ipList = explode(',', $string_ip);
-        foreach ($ipList as $ip) {
-            if (preg_match($pattern_extract_host, $ip, $matches)) {
-                $ip = $matches[2];
+        $hostList = array();
+        $pattern_extract_host = '/^[htps:\/]*([a-z0-9.]+)[:0-9]*$/';
```

Comment:
@cgagnaire @lpinsivy sounds like this will fail if URL contains `-` character.
Perhaps you should update to the following ?
`$pattern_extract_host = '/^[htps:\/]*([a-z0-9.-]+)[:0-9]*$/';`

Reply:
@CPbN should be $pattern_extract_host = '/^[htps:\/]*([a-z0-9.\-]+)[:0-9]*$/'; right?

Reply:
`\` may not be needed as `-` is at the end of the bracket enclosure.